marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs

```diff
@@ -401,7 +401,7 @@ module.exports.updateBatch = module.amp(
     }
 
     if (batchStatus === "finished" || batchStatus === "finished_with_errors" || batchStatus === "failed") {
-      docObj.batch.timeEnded = fn.currentDateTime();
+      docObj.batch.timeEnded = fn.currentDateTime().add(xdmp.elapsedTime());
```

Comment:
We definitely want to get a test in place for this. We don't care about exact times, just that timeEnded is after timeStarted. 

A good way to test this is via a custom step that just does an xdmp.sleep(10) call in it; that ensures there's no way that these timestamps should be the same. 

Check with @srinathgit  about how to write tests; also see the "Writing Data Hub Tests" internal Wiki page.

Reply:
Thank you.  Will take a look